scripting/shavit-misc.sp

```diff
@@ -2085,6 +2092,92 @@ public Action Shotgun_Shot(const char[] te_name, const int[] Players, int numCli
 	return Plugin_Stop;
 }
 
+public Action EffectDispatch(const char[] te_name, const Players[], int numClients, float delay)
+{
+	int iEffectIndex = TE_ReadNum("m_iEffectName");
+	int nHitBox = TE_ReadNum("m_nHitBox");
+	char sEffectName[64];
+
+	GetEffectName(iEffectIndex, sEffectName, 64);
+
+	if(gB_NoBlood)
+	{
+		if(StrEqual(sEffectName, "csblood"))
+		{
+			return Plugin_Handled;
+		}
+
+		if(StrEqual(sEffectName, "ParticleEffect"))
+		{
+			char sParticleEffectName[64];
+			GetParticleEffectName(nHitBox, sParticleEffectName, 64);
+
+			if(StrEqual(sParticleEffectName, "impact_helmet_headshot") || StrEqual(sParticleEffectName, "impact_physics_dust"))
+			{
+				return Plugin_Handled;
+			}
+		}
+	}
+
+	return Plugin_Continue;
+}
+
+public Action WorldDecal(const char[] te_name, const Players[], int numClients, float delay)
+{
+	float vecOrigin[3];
+	int nIndex = TE_ReadNum("m_nIndex");
+	char sDecalName[64];
+
+	TE_ReadVector("m_vecOrigin", vecOrigin);
+	GetDecalName(nIndex, sDecalName, 64);
+    
+	if(gB_NoBlood)
+	{
+		if(StrContains(sDecalName, "decals/blood") == 0 && StrContains(sDecalName, "_subrect") != -1)
+		{
+			return Plugin_Handled;
+		}
+	}
+
+	return Plugin_Continue;
+}
+
+public int GetParticleEffectName(int index, char[] sEffectName, int maxlen)
+{
+	int table = INVALID_STRING_TABLE;
```

Comment:
I think this is better as a `static int`, same for `GetEffectName` and `GetDecalfName`.